### PR TITLE
Use glob pattern to include routing annotations

### PR DESCRIPTION
**Pull request by ebuildy**
``FileLoader::import`` method can return an array, this PR fix the error "Call to a member function addPrefix() on array".

Use case "I want to import routes from controllers in different directories"; given a ``route/annotation.yaml`` like this:

```
controllers:
    resource: ../../src/*/Controller/*Controller.php
    type: annotation
```

(``/*Controller.php`` is required to make the glob return something).

I dont know how to code this properly, that why I just copied/paste the code like this, but the idea is here.

| Q             | A
| ------------- | ---
| Branch?       | master for features / 2.7 up to 4.0 for bug fixes 
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? |no
| Tests pass?   | yes/no
| Fixed tickets | #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#...

### CODE REVIEW

File: src/Symfony/Component/Routing/Loader/YamlFileLoader.php [hunk 1]
```diff
@@ -152,29 +152,58 @@ protected function parseImport(RouteCollection $collection, array $config, $path
...
-        if (null !== $methods) {
-            $subCollection->setMethods($methods);
-        }
-        $subCollection->addDefaults($defaults);
-        $subCollection->addRequirements($requirements);
-        $subCollection->addOptions($options);
+        
+        if (is_array($subCollection)) {
+            /* @var $value RouteCollection */
+            foreach($subCollection as $value) {
```
Comment (iltar) [inline src/Symfony/Component/Routing/Loader/YamlFileLoader.php]: This creates a lot of duplicate code. What if you simply ensure it's always an array?

```php
if (!is_array($subCollection)) {
    $subCollection = array($subCollection);
}
```

Comment (ebuildy) [reply src/Symfony/Component/Routing/Loader/YamlFileLoader.php]: absolutely, I didnt know the best way to deal with, thanks you.

**Comment by nicolas-grekas** [comment]
Could you add a test case please? If this is a bug fix, it should target the lowest maintained branch where the bug exists (either 2.7, 2.8, 3.3 or 3.4). Any hint in that aspect?

**Comment by ebuildy** [comment]
Sure, I have taken a look on ``Tests`` folder but dont know really how to implement this, I got error:

`Cannot load resource "/src/Symfony/Component/Routing/Tests/Fixtures/ControllerClasses/Sub1/Sub1Controller.php". Make sure annotations are installed and enabled.`

**Comment by nicolas-grekas** [comment]
> fix the error "Call to a member function addPrefix() on array".

so, this is a bug fix, isn't it?

> Make sure annotations are installed and enabled

please push a test case that you'd like to make work and we will be able to help I guess

**Comment by nicolas-grekas** [comment]
Oh, please also ensure that other loaders don't have the same issue.

**Comment by nicolas-grekas** [comment]
Closing in favor of #26600, thank you for working on this anyway!

**Pull request closed without merging**